lib/concurrent-ruby/concurrent/timer_task.rb

```diff
@@ -152,8 +169,16 @@ class TimerTask < RubyExecutorService
     # Default `:execution_interval` in seconds.
     EXECUTION_INTERVAL = 60
 
-    # Default `:timeout_interval` in seconds.
-    TIMEOUT_INTERVAL = 30
+    # Maintain the interval between the end of one execution and the start of the next execution.
+    FIXED_DELAY = :fixed_delay
```

Comment:
As discussed, we see a third mode, which could be `fixed_rate_drop` which drops subsequent executions if the current execution violates the maximum time slice given the rate specified.

Reply:
Yeah that makes sense to me.